master/internal/api_token.go

```diff
@@ -36,18 +37,37 @@ func (a *apiServer) PostAccessToken(
 		return nil, status.Error(codes.PermissionDenied, err.Error())
 	}
 
-	tokenExpiration := token.DefaultTokenLifespan
-	if req.Lifespan != "" {
-		d, err := time.ParseDuration(req.Lifespan)
-		if err != nil || d < 0 {
-			return nil, status.Error(codes.InvalidArgument,
-				"Lifespan must be a Go-formatted duration string with a positive value")
+	defaultTokenLifespan, err := token.ParseLifespanDays(a.m.config.Security.Token.DefaultLifespanDays)
+	if err != nil {
+		return nil, status.Error(codes.Internal, "Failed to parse default lifespan")
+	}
+
+	maxTokenLifespan, err := token.ParseLifespanDays(a.m.config.Security.Token.MaxLifespanDays)
```

Comment:
we should either do this when we parse the config or make a getter that does it so users don't need to remember to call `ParseLifespanDays` to avoid the footgun.

Reply:
i'm going to do something similar with one of my prs; i'll send you an example.

Reply:
I have tried to create a getter function, let me know if this is what you were talking about